### ISSUE TITLE
BatchNorm of ResNet implementation initialized as zero

### ISSUE THREAD

**Issue by yzcjtr** [deepmind/sonnet#185]
Hi, I find the scale of the last batch norm of each block is always initialized as zero: https://github.com/deepmind/sonnet/blob/v2/sonnet/src/nets/resnet.py#L93. This looks strange to me. Why don't we just initialize it as 1 instead of 0? Thank you.

**Comment by LenaMartens** [comment]
Hi! The Sonnet ResNet implementation closely follows this TensorFlow implementation (which also initializes the last batch norm in the residual block to 0): https://github.com/tensorflow/tpu/blob/d35e48588cfdab24aaddd045e37d0e967166931f/models/official/resnet/resnet_model.py#L443.

As for why: for example the [Accurate, Large Minibatch SGD: Training ImageNet in 1 Hour](https://arxiv.org/pdf/1706.02677.pdf) paper mentions initializing the last batch norm to 0 for easier training. From section 5.1: `Setting γ = 0 in the last BN of each residual block causes the forward/backward signal initially to propagate through the identity shortcut of ResNets, which we found to ease optimization at the start of training.`

Does this answer your question?

**Comment by yzcjtr** [comment]
Thanks @LenaMartens! Very nice explanations. That clears all of my questions.
